client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.controller.js

```diff
@@ -0,0 +1,376 @@
+        this.model.number = 1;
+
+        return this.$q.all({
+            flavors: this.OvhApiCloudProjectFlavor.Lexi().query({ serviceName: this.serviceName }).$promise
+                .then(flavors => {
+                    this.flavors = flavors;
+                    const filteredFlavors = this.VirtualMachineAddService.getAugmentedFlavorsFilteredByType(flavors, this.model.imageType.type);
+                    this.enums.flavorsTypes = this.CloudFlavorService.constructor.getFlavorTypes(filteredFlavors);
+                    return filteredFlavors;
+                }),
+            hasVRack: this.VirtualMachineAddService.hasVRack(this.serviceName),
+            prices: this.promisePrices
+                .then(prices => (this.prices = prices))
+                .catch(this.ServiceHelper.errorHandler("cpcivm_add_step3_flavor_prices_ERROR")),
+            publicNetworks: this.promisePublicNetworks
+                .then(publicNetworks => (this.publicNetworks = publicNetworks))
+                .catch(() => (this.publicNetworks = []))
+        })
+            .then(({ flavors, hasVRack }) => {
+                // Get Private Networks asynchronously
+                this.state.hasVRack = hasVRack;
+                if (hasVRack) {
+                    this.getPrivateNetworks();
+                }
+
+                // Add price and quota info to each instance type
+                _.forEach(flavors, flavor => {
+                    this.CloudFlavorService.constructor.addPriceInfos(flavor, this.prices);
+                    this.CloudFlavorService.constructor.addOverQuotaInfos(flavor, this.quota, _.get(this.model, "imageId.minDisk", 0), _.get(this.model, "imageId.minRam", 0));
+                });
+
+                // Remove flavor without price (not in the catalog)
+                _.remove(flavors, flavor => _.isEmpty(_.get(flavor, "price.price.text", "")));
+
+                let filteredFlavors = this.VirtualMachineAddService.getFilteredFlavorsByRegion(flavors, this.model.region.microRegion.code);
+
+                // Remove flavors if OS has restricted
+                const restrictedFlavors = _.get(this.model, "imageId.flavorType") || [];
+                if (restrictedFlavors.length > 0) {
+                    filteredFlavors = _.filter(filteredFlavors, flavor => _.indexOf(restrictedFlavors, flavor.shortType) > -1);
+                }
+
+                this.groupedFlavors = this.VirtualMachineAddService.groupFlavorsByCategory(filteredFlavors, this.enums.flavorsTypes);
+            })
+            .catch(this.ServiceHelper.errorHandler("cpcivm_add_step3_flavors_ERROR"))
+            .finally(() => {
+                this.loaders.step3 = false;
+            });
+    }
+
+    isStep3Valid () {
+        return this.model.flavor != null && !_.isEmpty(this.model.name) && this.model.number > 0 && (!this.state.hasVRack || !_.isEmpty(this.model.networkId));
+    }
+
+    resetStep3 () {
+        _.set(this.model, "flavor", null);
```

Comment:
since this.model will always be defined
this.model.flavor = null.

this apply for network and number too

Reply:
and model

Reply:
done